Makefile

```diff
@@ -2,22 +2,48 @@
 
 PYDIRS=setup.py examples scripts tests troposphere
 
-test:
-	flake8 ${PYDIRS}
-	python setup.py test
-	black --check ${PYDIRS}
-	isort --check ${PYDIRS}
+help: ## show this message
+	@IFS=$$'\n' ; \
+	help_lines=(`fgrep -h "##" $(MAKEFILE_LIST) | fgrep -v fgrep | sed -e 's/\\$$//' | sed -e 's/##/:/'`); \
+	printf "%-30s %s\n" "target" "help" ; \
+	printf "%-30s %s\n" "------" "----" ; \
+	for help_line in $${help_lines[@]}; do \
+		IFS=$$':' ; \
+		help_split=($$help_line) ; \
+		help_command=`echo $${help_split[0]} | sed -e 's/^ *//' -e 's/ *$$//'` ; \
+		help_info=`echo $${help_split[2]} | sed -e 's/^ *//' -e 's/ *$$//'` ; \
+		printf '\033[36m'; \
+		printf "%-30s %s" $$help_command ; \
+		printf '\033[0m'; \
+		printf "%s\n" $$help_info; \
+	done
 
-spec:
-	curl -O https://d1uauaxba7bl26.cloudfront.net/latest/CloudFormationResourceSpecification.zip
-	rm -rf spec
-	mkdir spec
-	unzip -d spec CloudFormationResourceSpecification.zip
-	rm CloudFormationResourceSpecification.zip
+fix-black: ## automatically fix all black errors
+	@black ${PYDIRS}
 
-spec2:
-	curl -O --compressed https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json
-	/bin/echo -n "Downloaded version: " && jq .ResourceSpecificationVersion CloudFormationResourceSpecification.json
+fix-isort: ## automatically fix all isort errors
+	@isort ${PYDIRS}
+
+clean:
+	rm -rf ${p39dir} troposphere.egg-info
+
+lint: lint-flake8 ## run all linters
```

Comment:
Shouldn't this also run `lint-black` and `lint-isort`?

Reply:
> I don't think this restriction is necessary

If I were to add the other targets here before the codebase is updated to pass both black and isort checks, the action I added will always fail. This target is being used as the main linting entry point for the workflow and to provide an entry point for local linting.

I plan to have a few follow-up PRs to this one that will address this concern by pairing it with updates to the code to pass these checks. I this them out, for now, to cut reduce the diffs for an easier review.